aimmo-game/simulation/world_map.py

```diff
@@ -85,22 +98,77 @@ def pickup_cells(self):
         return (c for c in self.all_cells() if c.pickup)
 
     def is_on_map(self, location):
-        return (0 <= location.y < self.num_rows) and (0 <= location.x < self.num_cols)
+        try:
+            self._grid[location]
+        except KeyError:
+            return False
+        return True
 
     def get_cell(self, location):
-        if not self.is_on_map(location):
+        try:
+            return self._grid[location]
+        except KeyError:
+            # For backwards-compatibility, this throws ValueError
             raise ValueError('Location %s is not on the map' % location)
-        cell = self.grid[location.x][location.y]
-        assert cell.location == location, 'location lookup mismatch: arg={}, found={}'.format(location, cell.location)
-        return cell
+
+    def get_cell_by_coords(self, x, y):
+        return self.get_cell(Location(x, y))
+
+    def max_y(self):
+        max_y = 0
+        while True:
+            try:
+                self._grid[Location(0, max_y)]
+            except KeyError:
+                max_y -= 1
+                break
+            else:
+                max_y += 1
+        return max_y
```

Comment:
These methods of going till your out of the map and then re-adjusting by one seem a bit hacky.
I prefer your suggestion of maintaining the value of the variable when outer layers are added (andpossibly removed in the future)